memreps/memreps.py

```diff
@@ -71,12 +91,27 @@ def find_distiguishing(concept1, concept2, atoms):
             assert (y in concept2) == polarity
             return x, y
 
+def find_maximally_distinguishing(concept_gen, max_concepts=10, atoms_per_concept=1):
+    concepts = fn.take(max_concepts, concept_gen())
+    pairs = combinations(concepts, 2)
+    if len(concepts) == 2: # pairs doesn't seem to return a single pair when len is 2?
```

Comment:
I just tried this and it seemed to work?

```python
In [1]: from itertools import combinations

In [2]: combinations([1,2], 2)
Out[2]: <itertools.combinations at 0x7f0498bd72c0>

In [3]: list(combinations([1,2], 2))
Out[3]: [(1, 2)]
```

Reply:
yeah i tried the same but for some reason in the loop it did not work ...

Reply:
in the function itself, i mean